src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js

```diff
@@ -0,0 +1,272 @@
+import MenuItem from '../../../../components/MenuItem';
+import styles from '../../../../styles/styles';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import Text from '../../../../components/Text';
+import OfflineWithFeedback from '../../../../components/OfflineWithFeedback';
+import ConfirmModal from '../../../../components/ConfirmModal';
+import * as User from '../../../../libs/actions/User';
+import TextInput from '../../../../components/TextInput';
+import CONST from '../../../../CONST';
+import Icon from '../../../../components/Icon';
+import colors from '../../../../styles/colors';
+import Button from '../../../../components/Button';
+import * as ErrorUtils from '../../../../libs/ErrorUtils';
+import themeColors from '../../../../styles/themes/default';
+import FullScreenLoadingIndicator from '../../../../components/FullscreenLoadingIndicator';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Value of partner name */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** Date when login was validated */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+
+        /** Field-specific pending states for offline UI status */
+        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    /** Current user session */
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** passed via route /settings/profile/contact-methods/:contactMethod/details */
+            contactMethod: PropTypes.string,
+        }),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    loginList: {},
+    route: {
+        params: {
+            contactMethod: '',
+        },
+    },
+};
+
+class ContactMethodDetailsPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleDeleteModal = this.toggleDeleteModal.bind(this);
+        this.confirmDeleteAndHideModal = this.confirmDeleteAndHideModal.bind(this);
+        this.resendValidateCode = this.resendValidateCode.bind(this);
+        this.getContactMethod = this.getContactMethod.bind(this);
+        this.validateContactMethod = this.validateContactMethod.bind(this);
+
+        this.state = {
+            isDeleteModalOpen: false,
+            validateCode: '',
+        };
+    }
+
+    componentDidMount() {
+        if (this.doesContactMethodExist()) {
+            return;
+        }
+        Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
```

Comment:
I didn't consider it really, but sounds I wouldn't mind changing to showing a "not found" page... @Beamanator thoughts?

Do we have some guidelines where we say we have this preference?

I see that we recently did this for reports here: https://github.com/Expensify/App/pull/15499

Reply:
I think that's the general pattern that we should follow because anyone can navigate to any URL, and IMO it makes more sense to show that the page is not found vs immediately navigate the user to a different URL.

Reply:
> I think that's the general pattern that we should follow because anyone can navigate to any URL, and IMO it makes more sense to show that the page is not found vs immediately navigate the user to a different URL.

Ok, I'll update it to this

Reply:
![image](https://user-images.githubusercontent.com/87341702/226722096-b63d556c-5a75-4ccf-bb33-942008ce7ade.png)

Updated!